### ISSUE TITLE
using FaCC?

### ISSUE THREAD

**Issue by fabienjuif** [ptitFicus/naive-router#18]
I quote one of pitfalls:
> Route components pass path and query parameters as props to child components. Thus there is a risk of name conflict. The priority order is : 1) path parameters 2) query parameters 3) props

Did you think of adding FaCC support? Like
**if the children is an instance of a function (ie: not an element), then:**
```jsx
{children(pathParams, queryParams)}
```
You don't need to pass props since the lib user can use it

Here an example of usage:
```jsx
<Route path="/foo/{id}">
  {(pathParams, queryParams) => (
    <Child
      customProps="yo"
      id={pathParams.id}
      compareId={queryParams.id}
    />
  )}
</Route>
```

**Comment by fabienjuif** [comment]
This is implicitly said: I think the current API and the FaCC one can coexists

**Comment by ptitFicus** [comment]
That's a very good idea, I'll add support for this !

Thank you !

**Comment by ptitFicus** [comment]
Fixed by a661fb47fafe463f71bed56f5786285c476e3e10
